pkg/apis/security/v1alpha1/common_lifecycle.go

```diff
@@ -69,3 +69,20 @@ func (pbs *PolicyBindingStatus) MarkBindingFailure(reason, messageFormat string,
 func (pbs *PolicyBindingStatus) MarkBindingAvailable() {
 	policybindingCondSet.Manage(pbs).MarkTrue(PolicyBindingConditionReady)
 }
+
+// PropagateBindingStatus propagates the current binding status from another binding status.
+func (pbs *PolicyBindingStatus) PropagateBindingStatus(other *PolicyBindingStatus) {
+	if other == nil {
+		return
+	}
+	pbs.InitializeConditions()
```

Comment:
I think this shouldn't be done here. This initializeConditions is typically done in the reconciler. And I think you are doing it here already: https://github.com/google/knative-gcp/blob/c8af8ac3b3723180be52aa94d7487f4675ef8233/pkg/reconciler/security/istio/eventpolicybinding/eventpolicybinding.go#L54

Let's get this in, and you remove it in a follow up.

Reply:
Ack